src/qt/bitcoingui.cpp

```diff
@@ -98,7 +98,7 @@ BitcoinGUI::BitcoinGUI(interfaces::Node& node, const PlatformStyle *_platformSty
     if(enableWallet)
     {
         /** Create wallet frame and make it the central widget */
-        walletFrame = new WalletFrame(_platformStyle, this);
+        walletFrame = new WalletFrame(_platformStyle, this, this);
```

Comment:
1f913ac9a00b982c997e70603ba62eebbe4c9d03

Why set parent if setCentralWidget changes it?

Reply:
If a widget is not intended to be a window why create it as a window (parent == `nullptr`)?

Reply:
Come on, it's being set as the central widget in the next line.. a minimal change to remove the circular dependency is
```diff
--- a/src/qt/walletframe.cpp
+++ b/src/qt/walletframe.cpp
@@ -5,7 +5,6 @@
 #include <qt/walletframe.h>
 #include <qt/walletmodel.h>

-#include <qt/bitcoingui.h>
 #include <qt/walletview.h>

 #include <cassert>
@@ -14,7 +13,7 @@
 #include <QLabel>

 WalletFrame::WalletFrame(const PlatformStyle *_platformStyle, BitcoinGUI *_gui) :
-    QFrame(_gui),
+    QFrame(nullptr),
     gui(_gui),
     platformStyle(_platformStyle)
 {
```